### ISSUE TITLE
Improve printing of getter types

### ISSUE THREAD

**Issue by smimram** [savonet/liquidsoap#786]
Right now the type of `gstreamer.hls` is
```
(?id:string,?on_error:((string)->float),'a)->source(audio=2,video=1,midi=0)
where 'a is either string or ()->string
```
It would be nice to hide the technical details of getters and display something like:
```
(?id:string,?on_error:((string)->float),{string})->source(audio=2,video=1,midi=0)
```
which would be much more readable, i.e. write `{t}` for a getter of type `t`.

What do you think?

**Comment by toots** [comment]
Just seeing this. Not a big fan because it's confusing:
- A getter of type `'a` can be either a static value of type '`a` or a function of type `() -> `a`
- `{v}` is already syntactic sugar for `fun () -> v`.
Type displaying a type `{float}` you will get people confused since you are incorrectly equating the former with the later.

**Comment by toots** [comment]
Just seeing this. Not a big fan because it's confusing:
- A getter of type `'a` can be either a static value of type '`a` or a function of type `() -> `a`
- `{v}` is already syntactic sugar for `fun () -> v`.
Type displaying a type `{float}` you will get people confused since you are incorrectly equating the former with the later.

**Comment by toots** [comment]
I vote for: `'a | () -> 'a`

**Comment by smimram** [comment]
Compare
```
(?id:string,?on_error:((string)->float),{string})->source(audio=2,video=1,midi=0)
```
with
```
(?id:string,?on_error:((string)->float),string | () -> string)->source(audio=2,video=1,midi=0)
```
Don't you find the first more readable?

The rationale is that most people will take for granted that `{string}` is roughly `string`. Moreover, I don't think the confusion with the same notation on values is harmful, on the contrary.

But there is still time to revert this.

**Comment by smimram** [comment]
This was finally merged in 0f98a13b480ec2fd0291aa57e118dd0205a47973.
